[feat] Do not allow chaining of the `-c` option; pass multiple paths as a colon separated list (#1125)
eth-cscs/reframe: docs/running.rst

   .. versionadded:: 2.12

.. note::

>    Using the command line ``-c`` or ``--checkpath`` multiple times is now deprecated and only the last option is going to be taken into account.

Reviewer: Also change the "is going to be taken into account." to "will be considered."

>    Multiple paths should be passed as a colon separated string to the above option:

Reviewer: ```suggestion
   Multiple paths should be passed instead as a colon separated list:
```

   .. code-block:: bash

>       ./bin/reframe -c /path/to/my/first/test.py:/path/to/my/second/test.py -r

Reviewer: Let the second path be a directory, so as to show that you can mix both.

>    .. versionchanged:: 2.22

Reviewer: ```suggestion
   .. versionchanged:: 3.0
```
